### ISSUE TITLE
Incorrect type definition for methods in XhrPlugin

### ISSUE THREAD

**Issue by amitnavindgi** [aws-observability/aws-rum-web#218]
Hi folks,

Thanks for developing `aws-rum-web`. It is a great library with a promising roadmap.

When I install `aws-rum-web` via `yarn` in a Vue application that uses Typescript 4.4 I get the following errors in dev server (`yarn run serve` which also results in a build failure via `yarn run build`. 

```
ERROR in /cw_rum/node_modules/aws-rum-web/dist/cjs/plugins/event-plugins/XhrPlugin.d.ts(61,19):
61:19 Property 'patches' in type 'XhrPlugin' is not assignable to the same property in base type 'MonkeyPatched<XMLHttpRequest, "open" | "send">'.
  Type '(MonkeyPatch<XMLHttpRequest, "open"> | { nodule: XMLHttpRequest; name: "send"; wrapper: () => (original: any) => (this: XMLHttpRequest) => void; })[]' is not assignable to type 'MonkeyPatch<XMLHttpRequest, "open" | "send">[]'.
    Type 'MonkeyPatch<XMLHttpRequest, "open"> | { nodule: XMLHttpRequest; name: "send"; wrapper: () => (original: any) => (this: XMLHttpRequest) => void; }' is not assignable to type 'MonkeyPatch<XMLHttpRequest, "open" | "send">'.
      Type 'MonkeyPatch<XMLHttpRequest, "open">' is not assignable to type 'MonkeyPatch<XMLHttpRequest, "open" | "send">'.
        Type '"open" | "send"' is not assignable to type '"open"'.
          Type '"send"' is not assignable to type '"open"'.
    59 |     constructor(config?: PartialHttpPluginConfig);
    60 |     protected onload(): void;
  > 61 |     protected get patches(): (MonkeyPatch<XMLHttpRequest, "open"> | {
       |                   ^
    62 |         nodule: XMLHttpRequest;
    63 |         name: "send";
    64 |         wrapper: () => (original: any) => (this: XMLHttpRequest) => void;
Version: typescript 4.4.4
```

I believe it is due to an incorrect type being inferred from [this line](https://github.com/aws-observability/aws-rum-web/blob/main/src/plugins/event-plugins/XhrPlugin.ts#L121).

I _think_ updating the [patches](https://github.com/aws-observability/aws-rum-web/blob/main/src/plugins/event-plugins/XhrPlugin.ts#L110) method in [XhrPlugin](https://github.com/aws-observability/aws-rum-web/blob/main/src/plugins/event-plugins/XhrPlugin.ts) to what's shown below should fix the issue. I was able to fix the issue by directly editing the type definition file to match the type used below.

```typescript
protected get patches() {
    return [
        {
            nodule: XMLHttpRequest.prototype,
            name: 'send' as const,
            wrapper: this.sendWrapper
        },
        {
            nodule: XMLHttpRequest.prototype,
            name: 'open' as const,
            wrapper: this.openWrapper
        }
    ] as MonkeyPatch<XMLHttpRequest, 'send' | 'open'>[];
}
```

This issue currently blocks the usage of `aws-rum-web` in Vue/Typescript applications. Let me know if that isn't the case and if it is due to something in my setup.

Packages of importance in my setup:

- Vue: 2.6.11
- Typescript: 4.4.4
- AWS Rum Web: 1.8.2
- Node: 12.16.3

Thanks!

**Comment by qhanam** [comment]
Thanks for the detailed issue @amitnavindgi ! Taking a look..

**Comment by qhanam** [comment]
We've published v1.8.3, which includes this patch. Thanks again for the contribution!

**Comment by amitnavindgi** [comment]
Thanks for the quick fix!
